c/src/arena_allocator.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed
+ * under the Apache 2.0 License. This product includes software developed at
+ * Datadog (https://www.datadoghq.com/).
+ *
+ * Copyright 2025-Present Datadog, Inc.
+ */
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+typedef struct {
+  uint8_t *buffer;
+  size_t size;
+  size_t offset;
+} plcs_arena_allocator;
```

Comment:
can this be opaque struct?
```suggestion
typedef struct plcs_arena_allocator;
```